feat: restructure `ibc` crate (#975)
cosmos/ibc-rs: ibc-clients/README.md

# IBC Clients

> This crate is top-level library re-exports implemented Inter-Blockchain
> Communication (IBC) light clients serves as a meta-crate, simplifying the
> process of importing and integrating various IBC clients into your blockchain.
> IBC is a distributed protocol that enables communication between distinct
> sovereign blockchains and IBC light clients track the consensus states and proof
> specs of external blockchains required to properly verify proofs against the
> client's consensus state.

Reviewer: ```suggestion
This top-level crate re-exports Inter-Blockchain
Communication (IBC) implementations of light clients. It serves as a meta-crate, simplifying the
process of importing and integrating various IBC clients into your blockchain.
IBC is a distributed protocol that enables communication between distinct
sovereign blockchains and IBC light clients track the consensus states and proof
specs of external blockchains, which are required to properly verify proofs against the
client's consensus state.
```
